pkg/controller/authentication/configmap.go

```diff
@@ -100,22 +101,29 @@ func (r *ReconcileAuthentication) handleConfigMap(instance *operatorv1alpha1.Aut
 		}
 	} else {
 		labels := currentConfigMap.Labels
-		ownerRefs := currentConfigMap.OwnerReferences
-		var ownRef string
-		for _, ownRefs := range ownerRefs {
-			ownRef = ownRefs.Kind
-		}
 
 		if labels != nil {
 			value, ok := labels["app"]
-			if ok && value == "auth-idp" && ownRef == "Authentication" {
+			if ok && value == "auth-idp" && ctrlCommon.IsControllerOf(instance, currentConfigMap) {
 				reqLogger.Info("ibmcloud-cluster-info Configmap is already created by IM operator", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", "ibmcloud-cluster-info")
-			} else if ok && value == "management-ingress" && ownRef == "ManagementIngress" {
+			} else if ok && value == "management-ingress" && ctrlCommon.GetControllerKind(currentConfigMap) == "ManagementIngress" {
 				reqLogger.Info("Configmap is already created by managementingress , IM installation may not proceed further until the configmap is removed", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", "ibmcloud-cluster-info")
 				*needToRequeue = true
 				return
 			} else {
-				reqLogger.Info("Can't determine the configmap ownership , IM installation may not proceed further until the configmap is removed", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", "ibmcloud-cluster-info")
+				reqLogger.Info("ConfigMap is not owned by the current Authentication or a ManagementIngress; will attempt to become controller", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", "ibmcloud-cluster-info")
+				if err = controllerutil.SetControllerReference(instance, currentConfigMap, r.client.Scheme()); err != nil {
+					reqLogger.Error(err, "Could not become the controller of the ConfigMap; it may need to be removed", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", "ibmcloud-cluster-info")
+					return
+				}
+				currentConfigMap.Labels["app"] = "auth-idp"
```

Comment:
Is this label correct?

Reply:
Based upon what installs on the cluster today in v3, yes.